### ISSUE TITLE
mask is ignored when assigning bool array to uint8 variable

### ISSUE THREAD

**Issue by srepmub** [Unidata/netcdf4-python#1271]
hi! the following gives a confusing result:

```
import netCDF4 as nc
import numpy as np
import numpy.ma as ma

ds = nc.Dataset('lal.nc', "w")

dim = ds.createDimension('time', 48)

var = ds.createVariable('blaat', np.uint8, ('time',),
    zlib=True, complevel=4, shuffle=True, fletcher32=True,
    fill_value=240)

mask = np.full((48,), False)
for x in range(30):
    mask[x] = True

mama = ma.array(np.full((48,), True), mask=mask)
#mama = ma.array(np.zeros((48,)), mask=mask)

print('masked elems', np.count_nonzero(mama.mask))

var[:] = mama
print('created var', var)

print('masked output', np.count_nonzero(var[:].mask))
```

leading to the following output:

```
masked elems 30
created var <class 'netCDF4._netCDF4.Variable'>
uint8 blaat(time)
    _FillValue: 240
unlimited dimensions: 
current shape = (48,)
filling on
masked output 0
```

so in the end, nothing is masked, even though a mask is passed with 30 masked elements..

interestingly, when swapping the commented line with the above (so passing ints instead of bools as data), it does work fine.

so it appears that when converting from bool to uint8, the mask gets lost..?

**Comment by jswhit** [comment]
Confirmed.  It works as intented however, if you cast the masked array to uint8 before assigning to the unit8 variable via
```
mama = mama.astype(np.uint8)
```

**Comment by jswhit** [comment]
potential fix n PR #1272

**Comment by srepmub** [comment]
awesome, thanks for the fast reaction!

**Comment by jswhit** [comment]
closed by PR #1272
